sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -422,7 +420,7 @@ function getValueOrFunctionResult(value: undefined | string | ((defaultValue: st
   return result;
 }
 
-function createDefaultRequestPolicyFactories(credentials: ServiceClientCredentials | RequestPolicyFactory | undefined, options: ServiceClientOptions): RequestPolicyFactory[] {
+function createDefaultRequestPolicyFactories(credentials: RequestPolicyFactory | undefined, options: ServiceClientOptions): RequestPolicyFactory[] {
```

Comment:
This is only called from one place, where we make sure that we dont pass `undefined`. So why is `undefined` supported here as a type?

Also, now that we don't support passing of the credentials directly, maybe rename the parameter to `requestPolicyFactory`?

Then, if the parameter is always guaranteed to be not `undefined` and of type `RequestPolicyFactory`, then just directly use `factories.push(credentials)` instead of the 
```javascript
if (credentials) {
    if (isRequestPolicyFactory(credentials)) {
       factories.push(credentials)
    }
}
```

Reply:
We haven't actually made sure that it isn't `undefined` at this point, we've only checked that if `credentials` isn't null then it's one of the supported types, otherwise we throw an exception.  However, I do think I can simplify all this logic using your suggestions, will give that a shot.